src/App.tsx

```diff
@@ -2,71 +1,97 @@
+import React, { useState, useMemo } from 'react';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
 
-export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+type Props = {
+  onDebounce?: (delay: number) => void;
+  onSelected?: (person: Person[] | null) => void;
+  debounceDelay?: number;
+};
+
+export const App: React.FC<Props> = ({
+  onDebounce,
+  onSelected,
+  debounceDelay = 300,
+}) => {
+  const [query, setQuery] = useState('');
+  const [people] = useState(peopleFromServer);
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+  const [isListVisible, setIsListVisible] = useState(true);
+
+  const debouncedSearch = useMemo(() => {
+    const timeoutId = setTimeout(() => {
+      onDebounce?.(debounceDelay);
+    }, debounceDelay);
+
+    return () => clearTimeout(timeoutId);
+  }, [debounceDelay, onDebounce]);
+
+  const filteredPeople = people.filter(person =>
+    person.name.toLowerCase().includes(query.toLowerCase()),
+  );
+
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+    setIsListVisible(true);
+    if (selectedPerson && selectedPerson.name !== event.target.value) {
+      setSelectedPerson(null);
+    }
+
+    debouncedSearch();
+  };
+
+  const handlePersonSelect = (person: Person) => {
+    setSelectedPerson(person);
+    setQuery(person.name);
+    setIsListVisible(false);
+    onSelected?.([person]);
```

Comment:
The `onSelected` callback is expected to receive a single `Person` object or `null`, but it is currently being called with an array containing a single `Person`. Update the call to `onSelected` to pass the `person` directly.